internal/prometheusbp/high_watermark.go

```diff
@@ -0,0 +1,111 @@
+package prometheusbp
+
+import (
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// HighWatermarkValue implements an int64 gauge with high watermark value.
+type HighWatermarkValue struct {
+	lock sync.RWMutex
```

Comment:
```suggestion
	mu sync.RWMutex
```
Is mu to refer to the lock more common?

Reply:
why do you want to use **mu** to refer to lock instead of using **lock** to refer to lock though :)

Reply:
For no good reason other than convention. I see it mostly being called `mu` in Go https://go.dev/tour/concurrency/9
Maybe because `lock.RLock()` reads a bit repetitive :shrug:

Reply:
`lock` is noun and `Lock`/`RLock` is verb :)